src/app/guards/auth.guard.ts

```diff
@@ -1,5 +1,19 @@
-import { CanActivateChildFn } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { CanActivateChild, Router } from '@angular/router';
 
-export const authGuard: CanActivateChildFn = (childRoute, state) => {
-  return true;
-};
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivateChild {
+  constructor(private router: Router) {}
+
+  canActivateChild(): boolean {
+    const token = sessionStorage.getItem('token');
```

Comment:
I suggest moving the storage logic to a service since, in the future, you will want to update the main storage to local storage/cockie or something else.